pkg/dockerregistry/server/auth.go

```diff
@@ -435,6 +427,37 @@ func verifyPruneAccess(ctx context.Context, c client.SelfSubjectAccessReviewsNam
 	return nil
 }
 
+func verifyImageStreamAccess(ctx context.Context, namespace, imageRepo, verb string, c client.SelfSubjectAccessReviewsNamespacer) error {
+	return verifyWithSAR(ctx, "imagestreams/layers", namespace, imageRepo, verb, c)
+}
+
+func verifyImageSignatureAccess(ctx context.Context, namespace, imageRepo string, c client.SelfSubjectAccessReviewsNamespacer) error {
+	return verifyWithSAR(ctx, "imagesignatures", namespace, imageRepo, "create", c)
+}
+
+func verifyPruneAccess(ctx context.Context, c client.SelfSubjectAccessReviewsNamespacer) error {
+	return verifyWithGlobalSAR(ctx, "images", "delete", c)
+}
+
+func verifyMetricsAccess(ctx context.Context, metrics configuration.Metrics, token string, c client.SelfSubjectAccessReviewsNamespacer) error {
+	if !metrics.Enabled {
+		return ErrOpenShiftAccessDenied
+	}
+
+	if len(metrics.Secret) > 0 {
+		if metrics.Secret != token {
+			return ErrOpenShiftAccessDenied
+		}
+		return nil
+	}
+
+	if err := verifyWithGlobalSAR(ctx, "metrics", "get", c); err != nil {
```

Comment:
`metrics` aren't a resource in the `image.openshift.io` API group... might make more sense as a subresource (e.g. `images/metrics`) or as a nonResourceURL (e.g. `/metrics`)? @smarterclayton @sttts, do you know what are we checking for metrics permission for non-apiserver components?

Reply:
@liggitt I assumed that it would be created role like:

https://github.com/openshift/origin/blob/8f44e6600770996be1acc29608b25e9d725d65eb/examples/prometheus/prometheus.yaml#L67-L77